web/packages/teleterm/src/services/config/appConfigSchema.ts

```diff
@@ -63,9 +75,30 @@ export const createAppConfigSchema = (platform: Platform) => {
       .describe(
         '`auto` uses modern ConPTY system if available, which requires Windows 10 (19H1) or above. Set to `winpty` to use winpty even if ConPTY is available.'
       ),
+    'terminal.shell': z
+      .string()
+      .default(settings.defaultOsShellId)
+      .describe(
+        'A default terminal shell. It is best to configure it through UI (right click on a terminal tab > Default Shell).'
+      )
+      .refine(
+        configuredShell =>
+          settings.availableShells.some(
+            shell => shell.id === configuredShell || CUSTOM_SHELL_ID
+          ),
+        configuredShell => ({
+          message: `Cannot find the shell "${configuredShell}". Available options are: [${settings.availableShells.map(({ id }) => id).join(', ')}]. Using platform default (${settings.defaultOsShellId}).`,
+        })
```

Comment:
When does this get evaluated? I set `terminal.shell` to a bogus value and I don't see this message anywhere.

Reply:
This is always true 😶‍🌫️
```ts
shell => shell.id === configuredShell || CUSTOM_SHELL_ID
```
I fixed this and simplified the condition.